packages/js/src/cache/notifications-cache.ts

```diff
@@ -0,0 +1,212 @@
+import { InMemoryCache } from './in-memory-cache';
+import type { Cache } from './types';
+import type { ListNotificationsArgs, ListNotificationsResponse, Notification } from '../notifications';
+import { NotificationEvents, NotificationsEvents, NovuEventEmitter } from '../event-emitter';
+import type { NotificationFilter } from '../types';
+
+const getFilterKey = ({ tags, read, archived }: Pick<ListNotificationsArgs, 'tags' | 'read' | 'archived'>): string => {
+  return JSON.stringify({ tags, read, archived });
+};
+
+const getPageKey = ({ limit, offset, after }: Pick<ListNotificationsArgs, 'limit' | 'offset' | 'after'>): string => {
+  return JSON.stringify({ limit, offset, after });
+};
+
+const singleNotificationEvents: NotificationEvents[] = [
+  'notification.read.pending',
+  'notification.read.resolved',
+  'notification.unread.pending',
+  'notification.unread.resolved',
+  'notification.archive.pending',
+  'notification.archive.resolved',
+  'notification.unarchive.pending',
+  'notification.unarchive.resolved',
+  'notification.complete_action.pending',
+  'notification.complete_action.resolved',
+  'notification.revert_action.pending',
+  'notification.revert_action.resolved',
+];
+
+const multipleNotificationEvents: NotificationsEvents[] = [
+  'notifications.read_all.pending',
+  'notifications.read_all.resolved',
+  'notifications.archive_all.pending',
+  'notifications.archive_all.resolved',
+  'notifications.archive_all_read.pending',
+  'notifications.archive_all_read.resolved',
+];
```

Comment:
These are all the events that the cache is listening to. The `pending` events send the optimistically updated data, the `resolved` events send the data from the server.